### ISSUE TITLE
API feedback

### ISSUE THREAD

**Issue by BitBarrel** [rive-app/rive-unity#9]
In general, the API is way too convoluted. There is too much coding required to get it to work from scratch without using the example scripts.

A few suggestions.

-Put all the renderqueue, command buffer, and camera code in a Rive.Setup() function.

-Setting triggers, bool, numbers, should not require making custom structs to set the data. It should be handled by the API instead. For example, this should work:
m_file.artboard(0).stateMachine(0).setNumber("test") = 50.0f;

-There should be no need to cast getNumber, because it is known that the number is a float. Same with a bool. This should work:
float test = m_file.artboard(0).stateMachine(0).getNumber("test");

-Artboards and state machines should have a name (string) property so that whatever is in the index can be checked if that is the correct one.

-An artboard and state machine index should have a function to get it with a string (name).

-In short, everything should have both a name (string) and an index (int) so it can both be found easily (slow), and referenced quickly (fast).

**Comment by BitBarrel** [comment]
[HayesGordon](https://github.com/HayesGordon)

**Comment by damzobridge** [comment]
Thanks again for the great feedback! For new users, closing this issue as the RivePanel/Widget components are now the recommended way to work with the Rive package in Unity.
